ontologytimemachine/utils/utils.py

```diff
@@ -0,0 +1,174 @@
+from proxy.http.parser import HttpParser, httpParserTypes
+from requests.exceptions import SSLError, Timeout, ConnectionError, RequestException
+from http.client import responses
+from urllib.parse import urlparse
+import logging
+import requests
+
+
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+
+dbpedia_api = 'https://archivo.dbpedia.org/download'
+
+
+ontology_types = [
+    'application/turtle',               # Turtle (Terse RDF Triple Language)
+    'text/turtle',                      # Turtle (alternative media type)
+    'application/rdf+xml',              # RDF/XML
+    'text/rdf+xml',                     # RDF/XML (alternative media type)
+    'application/n-triples',            # N-Triples
+    'text/n-triples',                   # N-Triples (alternative media type)
+    'application/n-quads',              # N-Quads
+    'text/n-quads',                     # N-Quads (alternative media type)
+    'application/ld+json',              # JSON-LD (JSON for Linking Data)
+    'application/trig',                 # TriG
+    'application/sparql-results+json',  # SPARQL Query Results JSON
+    'application/json',                 # JSON (alternative for SPARQL Query Results JSON)
+    'application/sparql-results+xml',   # SPARQL Query Results XML
+    'text/xml'                          # XML (alternative for SPARQL Query Results XML)
+]
+
+
+format_mapping = {
+    'application/turtle': 'ttl',               # Turtle (Terse RDF Triple Language)
+    'text/turtle': 'ttl',                      # Turtle (alternative media type)
+}
+
+
+passthrough_status_codes_http = [
+    100, 101, 102, 103,
+    200,
+    300, 301, 302, 303, 304, 307, 308,
+    451,
+]
+
+
+def check_if_archivo_ontology_requested(request):
+    urls = []
+    with open('ontologytimemachine/utils/archivo_ontologies.txt', 'r') as file:
```

Comment:
this needs to become a todo for later
needs to be faster (in memory - not open a file on every request) and also updated
periodically